tests/Testing/RectorRuleShouldNotBeApplied/RectorRuleShouldNotBeAppliedTest.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Tests\Testing\RectorRuleShouldNotBeApplied;
+
+use Iterator;
+use PHPUnit\Framework\Attributes\DataProvider;
+use Rector\Exception\Configuration\RectorRuleShouldNotBeAppliedException;
+use Rector\Testing\PHPUnit\AbstractRectorTestCase;
+
+final class RectorRuleShouldNotBeAppliedTest extends AbstractRectorTestCase
+{
+    #[DataProvider('provideData')]
+    public function test(string $filePath): void
+    {
+        $this->expectException(RectorRuleShouldNotBeAppliedException::class);
```

Comment:
To test the new functionality we run a rule that returns a Node even though it does not change it and then we confirm that we catch this case and throw an exception informing us of the problem